Common/src/Storage/IResultTable.cs

```diff
@@ -195,3 +198,38 @@ public interface IResultTable : IInitializable
   public record ChangeResultOwnershipRequest(IEnumerable<string> Keys,
                                              string              NewTaskId);
 }
+
+public static class IResultTableExt
+{
+  /// <summary>
+  ///   Get the result from its id
+  /// </summary>
+  /// <param name="resultTable"></param>
+  /// <param name="sessionId">id of the session containing the result</param>
+  /// <param name="key">id of the result to be retrieved</param>
+  /// <param name="cancellationToken">Token used to cancel the execution of the method</param>
+  /// <returns>
+  ///   Result metadata from the database
+  /// </returns>
+  public static async Task<Result> GetResult(this IResultTable resultTable,
```

Comment:
I think it is better to put this in the interface. If an adapter can be optimized in this case, it will be able to override it.

Reply:
I wanted to reduce code duplication but ok

Reply:
There is no code duplication if you put it in the interface (same file).

Reply:
this is the interface file